source/base/mpi.cc

```diff
@@ -386,136 +386,157 @@ namespace Utilities
         }
 
 #  if DEAL_II_MPI_VERSION_GTE(3, 0)
-      // check if destinations contain duplicates
-      auto destinations_unique = destinations;
-      std::sort(destinations_unique.begin(), destinations_unique.end());
-      destinations_unique.erase(std::unique(destinations_unique.begin(),
-                                            destinations_unique.end()),
-                                destinations_unique.end());
-
-      if (Utilities::MPI::min<unsigned int>(destinations.size() ==
-                                              destinations_unique.size(),
-                                            mpi_comm))
+
+      // Have a little function that checks if destinations provided
+      // to the current process are unique:
+      const bool my_destinations_are_unique = [destinations]() {
+        std::vector<unsigned int> my_destinations = destinations;
+        const unsigned int        n_destinations  = my_destinations.size();
+        std::sort(my_destinations.begin(), my_destinations.end());
+        my_destinations.erase(std::unique(my_destinations.begin(),
+                                          my_destinations.end()),
+                              my_destinations.end());
+        return (my_destinations.size() == n_destinations);
+      }();
+
+      // If all processes report that they have unique destinations,
+      // then we can short-cut the process using a consensus algorithm (which
+      // is implemented only for the case of unique destinations):
+      if (Utilities::MPI::min((my_destinations_are_unique ? 1 : 0), mpi_comm) ==
+          1)
         {
           ConsensusAlgorithms::AnonymousProcess<char, char> process(
             [&]() { return destinations; });
           ConsensusAlgorithms::NBX<char, char> consensus_algorithm(process,
                                                                    mpi_comm);
           return consensus_algorithm.run();
         }
+      else
+        // If that was not the case, we need to use the remainder of the code
+        // below, i.e., just fall through the else branch here
+        {}
```

Comment:
I would prefer to just omit this.
```suggestion
```

Reply:
Suggested changes spanning multiple lines, finally! :smiley: